x-pack/plugins/security_solution/public/management/pages/policy/view/event_filters/flyout/policy_event_filters_flyout.test.tsx

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import React from 'react';
+import uuid from 'uuid';
+import { getFoundExceptionListItemSchemaMock } from '../../../../../../../../lists/common/schemas/response/found_exception_list_item_schema.mock';
+import { getExceptionListItemSchemaMock } from '../../../../../../../../lists/common/schemas/response/exception_list_item_schema.mock';
+import {
+  AppContextTestRender,
+  createAppRootMockRenderer,
+} from '../../../../../../common/mock/endpoint';
+import { EndpointDocGenerator } from '../../../../../../../common/endpoint/generate_data';
+import { PolicyData } from '../../../../../../../common/endpoint/types';
+import { getPolicyEventFiltersPath } from '../../../../../common/routing';
+import { eventFiltersListQueryHttpMock } from '../../../../event_filters/test_utils';
+import { PolicyEventFiltersFlyout } from './policy_event_filters_flyout';
+import { parseQueryFilterToKQL, parsePoliciesAndFilterToKql } from '../../../../../common/utils';
+import { SEARCHABLE_FIELDS } from '../../../../event_filters/constants';
+import {
+  FoundExceptionListItemSchema,
+  UpdateExceptionListItemSchema,
+} from '@kbn/securitysolution-io-ts-list-types';
+
+const endpointGenerator = new EndpointDocGenerator('seed');
+const getDefaultQueryParameters = (customFilter: string | undefined = '') => ({
+  path: '/api/exception_lists/items/_find',
+  query: {
+    filter: customFilter,
+    list_id: ['endpoint_event_filters'],
+    namespace_type: ['agnostic'],
+    page: undefined,
+    per_page: 100,
+    sort_field: undefined,
+    sort_order: undefined,
+  },
+});
+const emptyList = {
+  data: [],
+  page: 1,
+  per_page: 10,
+  total: 0,
+};
+
+const getCleanedExceptionWithNewTags = (
+  exception: UpdateExceptionListItemSchema,
+  testTags: string[],
+  policy: PolicyData
+) => {
+  const exceptionToUpdateCleaned = {
+    ...exception,
+    tags: [...testTags, `policy:${policy.id}`],
+  };
+  // Clean unnecessary fields for update action
```

Comment:
is this the same code as the service file? if so, maybe create a utility that does this (TA has a similar one I think) and reuse it.